### DISCUSSION TITLE
Avalonia TextBox input ime mode issue

### DISCUSSION THREAD

**Discussion by tund3010** [AvaloniaUI/Avalonia#7826] [Q&A]
I am building a Avalonia desktop app to run on Ubuntu system. My app must support inputting Vietnamese (ibus). There is an issue that preedit content is displayed out-side the TextBox
![dWufw](https://user-images.githubusercontent.com/5442114/158837502-238f7a3a-f7b2-4912-bed0-531ba1228868.png)
How do I make the preedit content displays inside TextBox? Please help!

**Comment by Gillibald** [comment]
Are you using current master?

**Reply by tund3010** [reply]
Yes, I am using the latest version 0.10.13

**Comment by kekekeks** [comment]
We don't currently support preedit embedded into the textbox itself. It's a planned feature, but no ETA yet.

**Reply by tund3010** [reply]
Is there any alternative control to support embedded preedit? Or could you please give some instructions to implement embedded preedit in textbox by myselft? Thanks.
